server/src/main/java/org/elasticsearch/cluster/coordination/PreVoteCollector.java

```diff
@@ -105,6 +108,12 @@ private PreVoteResponse handlePreVoteRequest(final PreVoteRequest request) {
         final DiscoveryNode leader = state.v1();
         final PreVoteResponse response = state.v2();
 
+        //TODO verify if the placement makes sense
+        if (nodeHealthService.getHealth() == NodeHealthService.Status.UNHEALTHY) {
+            logger.warn("Reject offering pre-vote as all paths are not writable");
```

Comment:
A warning here isn't helpful, we should be logging the failure elsewhere so this will simply result in confusion.

```suggestion
            logger.debug("Reject offering pre-vote as all paths are not writable");
```

Also, we have this generic `NodeHealthService` but the log message is very specific: `all data paths are not writable`. Maybe the `NodeHealthService` should describe the problem rather than returning a simple boolean. Possibly makes sense to construct the exception within the `NodeHealthService`.